k8s/job.go

```diff
@@ -33,9 +33,10 @@ func ApplyK8sSystemJob(jobYaml, kubeConfigPath string, k8sWrapTransport WrapTran
 	if err != nil {
 		return err
 	}
-	// if the addon configMap is updated, or the previous job is not completed,
-	// I will remove the existing job first, if any
-	if addonUpdated || (jobStatus.Created && !jobStatus.Completed) {
+
+	// The safest way to ensure add-ons are in sync with the cluster.yaml is to always re-run the job and let
+	// kubectl apply determine what if any changes need to be made.
+	if addonUpdated || jobStatus.Created || jobStatus.Completed {
 		logrus.Debugf("[k8s] replacing job %s.. ", job.Name)
```

Comment:
nit: can we change the debug message here if we are always going to run kubectl apply, so that its not misleading

Reply:
I updated it from:
> logrus.Debugf("[k8s] replacing job %s.. ", job.Name)
to:
> logrus.Debugf("[k8s] deleting job %s so that it can be recreated.. ", job.Name)

Let me know if that works.